0.md

```diff
@@ -0,0 +1,322 @@
+# First-class Module and ModuleSource
+
+## Synopsis
+
+Provide first-class `Module` and `ModuleSource` constructors and extend dynamic
+import to operate on `Module` instances.
+
+A `ModuleSource` represents the result of compiling EcmaScript module source
+text.
+A `Module` instance represents the lifecycle of a EcmaScript module and allows
```

Comment:
I’ll add a note to the design considerations. @littledan and I tentatively agreed to float this idea since we think the waveform of the module harmony proposals is likely to collapse in the pleasing state where `(module {}) instanceof Module`.